src/numerics/petsc_vector.C

```diff
@@ -1384,8 +1384,46 @@ void PetscVector<T>::_restore_array() const
     }
 }
 
+template <typename T>
+std::unique_ptr<NumericVector<T>>
+PetscVector<T>::get_subvector(const std::vector<numeric_index_type> & rows)
+{
+  // Construct index set
+  WrappedPetsc<IS> parent_is;
+  auto ierr = ISCreateGeneral(this->comm().get(),
+                              cast_int<PetscInt>(rows.size()),
+                              numeric_petsc_cast(rows.data()),
+                              PETSC_USE_POINTER,
+                              parent_is.get());
+  LIBMESH_CHKERR(ierr);
 
+  Vec subvec;
+  ierr = VecGetSubVector(_vec, parent_is, &subvec);
```

Comment:
We allocate memory for storing this sub vector here, I guess?

Reply:
It may or it may not. Sometimes PETSc provides a direct view into the memory of the existing vector in which case there is no new memory allocation